core/funcs.go

```diff
@@ -852,6 +899,141 @@ func init() {
+			},
+		},
+		{
+			Name:             FUNC_CEIL,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslFloatT, RslIntT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				arg := f.args[0]
+				val := arg.value.RequireFloatAllowingInt(f.i, arg.node)
+				return newRslValues(f.i, f.callNode, math.Ceil(val))
+			},
+		},
+		{
+			Name:             FUNC_FLOOR,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslFloatT, RslIntT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				arg := f.args[0]
+				val := arg.value.RequireFloatAllowingInt(f.i, arg.node)
+				return newRslValues(f.i, f.callNode, math.Floor(val))
+			},
+		},
+		{
+			Name:             FUNC_MIN,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslListT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				arg := f.args[0]
+				list := arg.value.RequireList(f.i, arg.node)
+				if list.Len() == 0 {
+					f.i.errorf(f.callNode, "Cannot find minimum of empty list")
+				}
+
+				minVal := math.MaxFloat64
+				for idx, item := range list.Values {
+					val, ok := item.TryGetFloatAllowingInt()
+					if !ok {
+						f.i.errorf(arg.node, "%s() requires a list of numbers, got %q at index %d", FUNC_MIN, TypeAsString(item), idx)
+					}
+					minVal = math.Min(minVal, val)
+				}
+				return newRslValues(f.i, f.callNode, minVal)
+			},
+		},
+		{
+			Name:             FUNC_MAX,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 1,
+			ArgTypes:         [][]RslTypeEnum{{RslListT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				arg := f.args[0]
+				list := arg.value.RequireList(f.i, arg.node)
+				if list.Len() == 0 {
+					f.i.errorf(f.callNode, "Cannot find maximum of empty list")
+				}
+
+				maxVal := -math.MaxFloat64
+				for idx, item := range list.Values {
+					val, ok := item.TryGetFloatAllowingInt()
+					if !ok {
+						f.i.errorf(arg.node, "%s() requires a list of numbers, got %q at index %d", FUNC_MAX, TypeAsString(item), idx)
+					}
+					maxVal = math.Max(maxVal, val)
+				}
+				return newRslValues(f.i, f.callNode, maxVal)
+			},
+		},
+		{
+			Name:             FUNC_CLAMP,
+			ReturnValues:     ONE_RETURN_VAL,
+			RequiredArgCount: 3,
+			ArgTypes:         [][]RslTypeEnum{{RslFloatT, RslIntT}, {RslFloatT, RslIntT}, {RslFloatT, RslIntT}},
+			NamedArgs:        NO_NAMED_ARGS,
+			Execute: func(f FuncInvocationArgs) []RslValue {
+				// input is a number and a min and max
+				val, ok := f.args[0].value.TryGetFloatAllowingInt()
+				if !ok {
+					f.i.errorf(f.args[0].node, "%s() requires a number, got %q at index %d", FUNC_CLAMP, TypeAsString(f.args[0].value), 0)
+				}
+
+				minVal, ok := f.args[1].value.TryGetFloatAllowingInt()
```

Comment:
Similar to the RequiredArgCount, you don't need to call the Try functions here and check ok. The ArgTypes list above enforces that, when Execute is invoked, f.args you're given here is of the correct types.

So you can safely invoke RequireFloatAllowingInt, here :)